version: "3.7"
services:
  nginx:
    image: nginx:1.19.5
    networks:
      - gis_1ban_02_network
    volumes:
    - gis_1ban_02_static:/data/static/
    - gis_1ban_02_media:/data/media/
    - /home/kyeongseojeon/nginx.conf:/etc/nginx/nginx.conf
    ports:
    - 80:80

  django:
    image: gis_1ban_02_django:4
    networks:
      - gis_1ban_02_network
    secrets:
      - DJANGO_SECRET_KEY
      - MARIADB_USER
      - MARIADB_PASSWORD
    volumes:
    - gis_1ban_02_static:/home/gis_1ban_02/staticfiles/
    - gis_1ban_02_media:/home/gis_1ban_02/media/

  mariadb:
    image: mariadb:10.5
    networks:
      - gis_1ban_02_network
    volumes:
    - gis_mariadb:/var/lib/mysql
    secrets:
      - MARIADB_USER
      - MARIADB_PASSWORD
      - MARIADB_ROOT_PASSWORD
    environment:
      MARIADB_DATABASE: django
      MARIADB_USER_FILE: /run/secrets/MARIADB_USER
      MARIADB_PASSWORD_FILE: /run/secrets/MARIADB_PASSWORD
      MARIADB_ROOT_PASSWORD_FILE: /run/secrets/MARIADB_ROOT_PASSWORD

networks:
  gis_1ban_02_network:

volumes:
  gis_1ban_02_static:
  gis_1ban_02_media:
  gis_mariadb:

secrets:
  MARIADB_USER:
    external: true
  MARIADB_PASSWORD:
    external: true
  MARIADB_ROOT_PASSWORD:
    external: true
  DJANGO_SECRET_KEY:
    external: true
